frontend/src/components/Agreements/AgreementDetails/AgreementHistoryPanel.jsx

```diff
@@ -0,0 +1,173 @@
+import PropTypes from "prop-types";
+import { useGetAgreementHistoryByIdQuery } from "../../../api/opsAPI";
+import { historyData } from "../../../pages/agreements/details/data";
+import LogItem from "../../UI/LogItem";
+import { convertCodeForDisplay } from "../../../helpers/utils";
+
+const findObjectTitle = (historyItem) => {
+    if (historyItem.class_name === "BudgetLineItem") {
+        return historyItem.event_details.line_description;
+    } else {
+        return historyItem.event_details.name;
+    }
+};
+
+const summaryMessage = (historyItem) => {
+    const className = convertCodeForDisplay("className", historyItem.class_name);
+    const user_name = historyItem.created_by_user_full_name;
```

Comment:
should user_name be userName to be consistant camelCase where possible?

Reply:
done